src/device-manager/java/WeaveDeviceManager-JNI.cpp

```diff
@@ -3321,6 +3452,631 @@ WEAVE_ERROR GetClassRef(JNIEnv *env, const char *clsType, jclass& outCls)
+{
+    WDMClient *wdmClient = (WDMClient *)wdmClientPtr;
+
+    WeaveLogProgress(DeviceManager, "DeleteWDMClient() called");
+
+    if (wdmClient != NULL)
+    {
+        if (wdmClient->mpAppState != NULL)
+        {
+            env->DeleteGlobalRef((jobject)wdmClient->mpAppState);
+        }
+        wdmClient->Close();
+        delete wdmClient;
+    }
+}
+
+jlong Java_nl_Weave_DataManagement_WDMClientImpl_newDataSink(JNIEnv *env, jobject self, jlong wdmClientPtr, jobject resourceIdentifierObj, jlong profileId, jlong instanceId, jstring path)
+{
+    WEAVE_ERROR err = WEAVE_NO_ERROR;
+    long res = 0;
+    const char *pPathStr = NULL;
+    ResourceIdentifier resourceIdentifier;
+    GenericTraitUpdatableDataSink * dataSink = NULL;
+    WDMClient  *wdmClient = (WDMClient *)wdmClientPtr;
+
+    WeaveLogProgress(DeviceManager, "newDataSink() called");
+
+    err = J2N_ResourceIdentifier(env, resourceIdentifierObj, resourceIdentifier);
+    SuccessOrExit(err);
+
+    pPathStr = env->GetStringUTFChars(path, 0);
+    VerifyOrExit(pPathStr != NULL, err = WEAVE_ERROR_NO_MEMORY);
+
+    err = wdmClient->NewDataSink(resourceIdentifier, (uint32_t)profileId, (uint64_t)instanceId, pPathStr, dataSink);
+
+exit:
+    if (err == WEAVE_NO_ERROR && dataSink != NULL)
+    {
+        res = (long) dataSink;
+    }
+
+    if (pPathStr != NULL)
+    {
+        env->ReleaseStringUTFChars(path, pPathStr);
+    }
+
+    if (err != WEAVE_NO_ERROR && err != WDM_JNI_ERROR_EXCEPTION_THROWN)
+    {
+        ThrowError(env, err);
+    }
+
+    return res;
+}
+
+void Java_nl_Weave_DataManagement_WDMClientImpl_beginFlushUpdate(JNIEnv *env, jobject self, jlong wdmClientPtr)
+{
+    WEAVE_ERROR err = WEAVE_NO_ERROR;
+    WDMClient  *wdmClient = (WDMClient *)wdmClientPtr;
+
+    WeaveLogProgress(DeviceManager, "beginFlushUpdate() called");
+
+    pthread_mutex_lock(&sStackLock);
+    err = wdmClient->FlushUpdate((void *)"FlushUpdate", HandleWDMClientComplete, HandleWDMClientError);
+    pthread_mutex_unlock(&sStackLock);
+
+    if (err != WEAVE_NO_ERROR && err != WDM_JNI_ERROR_EXCEPTION_THROWN)
+    {
+        ThrowError(env, err);
+    }
+}
+
+void Java_nl_Weave_DataManagement_WDMClientImpl_beginRefreshData(JNIEnv *env, jobject self, jlong wdmClientPtr)
+{
+    WEAVE_ERROR err = WEAVE_NO_ERROR;
+    WDMClient  *wdmClient = (WDMClient *)wdmClientPtr;
+
+    WeaveLogProgress(DeviceManager, "beginRefreshData() called");
+
+    pthread_mutex_lock(&sStackLock);
+    err = wdmClient->RefreshData((void *)"RefreshData", HandleWDMClientComplete, HandleWDMClientError, NULL);
+    pthread_mutex_unlock(&sStackLock);
+
+    if (err != WEAVE_NO_ERROR && err != WDM_JNI_ERROR_EXCEPTION_THROWN)
+    {
+        ThrowError(env, err);
+    }
+}
+
+void Java_nl_Weave_DataManagement_GenericTraitUpdatableDataSinkImpl_init(JNIEnv *env, jobject self, jlong genericTraitUpdatableDataSinkPtr)
+{
+    GenericTraitUpdatableDataSink  *pDataSink = NULL;
```

Comment:
Here and the next several, there's two spaces between the type and the *

Reply:
thanks for this catch. fixing it.